api/rosetta/data_transaction_integration_test.go

```diff
@@ -502,13 +435,10 @@ func TestAPI_TransactionHandlesMalformedRequest(t *testing.T) {
 
 			t.Parallel()
 
-			req := httptest.NewRequest(http.MethodPost, "/block/transaction", bytes.NewReader(test.payload))
-			req.Header.Set(echo.HeaderContentType, test.mimeType)
-
-			rec := httptest.NewRecorder()
-			ctx := echo.New().NewContext(req, rec)
+			_, ctx, err := setupRecorder(transactionEndpoint, test.payload, test.prepare)
```

Comment:
Are there cases where prepare does smth else than to set the content type header? 🤔

Reply:
No, that's it. To be more precise, the content type is set to `application/json` automatically in the `setupRecorder()` function, and this function here is because we want to clear it for the test case where it's unset.

The other test cases have it basically only to keep the same interface and not have some special handling in the test case body. I.e. I didn't want to have

```go
if test.prepare != nil {
  _, ctx, error = setupRecorder(optionsEndpoint, test.payload, test.prepare)
} else {
  _, ctx, error = setupRecorder(optionsEndpoint, test.payload)
}
```

I also didn't want to have `setupRecorder` check for any `nil` options passed to it.